test/test_images/probe_helper/utils/liveness.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"context"
+	nethttp "net/http"
+
+	"go.uber.org/zap"
+	"knative.dev/pkg/logging"
+)
+
+// actionFunc represents a function which is called during a liveness probe. If
+// it returns a non-nil error, the probe is considered unsuccessful.
+type actionFunc func(context.Context) error
+
+// LivenessChecker executes each of the actionFuncs upon each liveness probe.
+type LivenessChecker struct {
+	actionFuncs []actionFunc
+}
+
+// AddActionFunc appends an actionFunc to the list of functions to be called
+// by the liveness checker during each liveness probe.
+func (c *LivenessChecker) AddActionFunc(f actionFunc) {
+	c.actionFuncs = append(c.actionFuncs, f)
+}
+
+// LivenessHandlerFunc returns the HTTP handler for probe helper liveness checks.
+func (c *LivenessChecker) LivenessHandlerFunc(ctx context.Context) nethttp.HandlerFunc {
+	return func(w nethttp.ResponseWriter, req *nethttp.Request) {
+		// Only accept liveness requests along the 'healthz' path
+		if req.URL.Path != "/healthz" {
+			logging.FromContext(ctx).Warnw("Invalid liveness check request path", zap.String("path", req.URL.Path))
+			w.WriteHeader(nethttp.StatusNotFound)
+			return
+		}
+		// Apply every actionFunc, do not interrupt when encountering an error
+		success := true
+		for _, f := range c.actionFuncs {
+			if err := f(ctx); err != nil {
+				logging.FromContext(ctx).Infow("Liveness check failed", zap.Error(err))
```

Comment:
nit: you can use multerr package to aggregate errors. See https://github.com/google/knative-gcp/blob/df0ee9b094c7885a98edf9d93f15dc60eafaa561/pkg/reconciler/broker/broker.go#L29